src/main/webapp/app/quiz/participate/quiz.component.html

```diff
@@ -11,20 +11,36 @@ <h2 [ngSwitch]="mode">
-        <jhi-multiple-choice-question *ngIf="question.type === MULTIPLE_CHOICE" [question]="question" [(selectedAnswerOptions)]="selectedAnswerOptions[question.id]" [fnOnSelection]="onSelectionChanged.bind(this)" [clickDisabled]="submission.submitted || remainingTimeSeconds < 0" [showResult]="showingResult" [forceSampleSolution]="mode === 'solution'" [questionIndex]="i + 1" [score]="questionScores[question.id]"></jhi-multiple-choice-question>
-        <jhi-drag-and-drop-question *ngIf="question.type === DRAG_AND_DROP" [question]="question" [(mappings)]="dragAndDropMappings[question.id]" [fnOnMappingUpdate]="onSelectionChanged.bind(this)" [clickDisabled]="submission.submitted || remainingTimeSeconds < 0" [showResult]="showingResult" [forceSampleSolution]="mode === 'solution'" [questionIndex]="i + 1" [score]="questionScores[question.id]"></jhi-drag-and-drop-question>
-        <jhi-short-answer-question *ngIf="question.type === SHORT_ANSWER" [question]="question" [(submittedTexts)]="shortAnswerSubmittedTexts[question.id]" [fnOnSubmittedTextUpdate]="onSelectionChanged.bind(this)" [clickDisabled]="submission.submitted || remainingTimeSeconds < 0" [showResult]="showingResult" [forceSampleSolution]="mode === 'solution'" [questionIndex]="i + 1" [score]="questionScores[question.id]"></jhi-short-answer-question>
+        <jhi-multiple-choice-question id="question{{i}}" *ngIf="question.type === MULTIPLE_CHOICE" [question]="question" [(selectedAnswerOptions)]="selectedAnswerOptions[question.id]" [fnOnSelection]="onSelectionChanged.bind(this)" [clickDisabled]="submission.submitted || remainingTimeSeconds < 0" [showResult]="showingResult" [forceSampleSolution]="mode === 'solution'" [questionIndex]="i + 1" [score]="questionScores[question.id]"></jhi-multiple-choice-question>
+        <jhi-drag-and-drop-question id="question{{i}}"  *ngIf="question.type === DRAG_AND_DROP" [question]="question" [(mappings)]="dragAndDropMappings[question.id]" [fnOnMappingUpdate]="onSelectionChanged.bind(this)" [clickDisabled]="submission.submitted || remainingTimeSeconds < 0" [showResult]="showingResult" [forceSampleSolution]="mode === 'solution'" [questionIndex]="i + 1" [score]="questionScores[question.id]"></jhi-drag-and-drop-question>
+        <jhi-short-answer-question id="question{{i}}" *ngIf="question.type === SHORT_ANSWER" [question]="question" [(submittedTexts)]="shortAnswerSubmittedTexts[question.id]" [fnOnSubmittedTextUpdate]="onSelectionChanged.bind(this)" [clickDisabled]="submission.submitted || remainingTimeSeconds < 0" [showResult]="showingResult" [forceSampleSolution]="mode === 'solution'" [questionIndex]="i + 1" [score]="questionScores[question.id]"></jhi-short-answer-question>
     </div>
 </div>
 <div class="quiz-footer" *ngIf="quizExercise">
     <div class="container">
         <div class="quiz-footer-content">
             <div>
-                <div *ngIf="quizExercise.questions">
+                <div class="stepwizardquiz">
```

Comment:
the code below looks very complex, because we have the three cases of DD, MC and SA several times. I suggest you refactor this so that you have only one distinction of the question.type at the beginning (directly after the for loop) and then you set the color without boolean operators and query only one corresponding variables selectedAnswerOptions, dragAndDropMappings and shortAnswerSubmittedTexts depending on the question type. Another option might be to extract methods for these queries to make them easier